test/partitioning.jl

```diff
@@ -13,7 +13,11 @@
     p = partition(grid, UniformPartition(3, false))
     @test setify(indices(p)) == setify([[1,2,3], [4,5,6], [7,8,9]])
     p = partition(grid, UniformPartition(3))
-    @test setify(indices(p)) == setify([[8,6,9], [4,1,7], [2,3,5]])
+    if VERSION < v"1.7"
```

Comment:
Can we compare with v1.6 somehow? A comparison that honors semantic versioning.

Reply:
I think you can do something like `major(VERSION) == 1 && minor(VERSION) == 6`

Reply:
It seems we are supporting `v1.5` (see [Project.toml](https://github.com/JuliaGeometry/Meshes.jl/blob/d35561d21301039f142c0448e7f2eacd8c368d02/Project.toml)), so I think it's better to keep `VERSION < v"1.7"` here.

Reply:
Maybe we will drop support Julia v1.5 in the future, but I think It's better to avoid unintentional errors with `< v"1.7.0"`.

Reply:
Yes, I think we should only support LTS and the latest release. It is too much on us maintainers to keep the project working across multiple versions that none of us are using anymore. I am planning to do a major cleanup in all Project.toml and CI of a bunch of packages to only support LTS and latest, that is why I think we should drop v1.5 from here as well.

Reply:
It is a fresh restart to officially state what versions are supported. We've been lazy about it but as the package matures towards a v1.0 we will need to adopt some rule like that.

Reply:
>Yes, I think we should only support LTS and the latest release. It is too much on us maintainers to keep the project working across multiple versions that none of us are using anymore. I am planning to do a major cleanup in all Project.toml and CI of a bunch of packages to only support LTS and latest, that is why I think we should drop v1.5 from here as well.

I agree with that:+1:

>A comparison that honors semantic versioning.

Julia is semantically versioned, so I'm not sure why we should avoid `VERSION < v"1.7"`.
It seems `VERSION < v"1.7"` is easier to read, fewer characters, and more meaningful than `major(VERSION) == 1 && minor(VERSION) == 6`, I guess.

Is it okay to just adding a comment like `# The random number generation algorithm has been changed since Julia 1.7.` ?

Reply:
Sounds good, let's merge it and then we can sort out a more explicit documentation about the supported versions of the language.